solr/core/src/test/org/apache/solr/search/stats/TestDistribIDF.java

```diff
@@ -263,4 +267,51 @@ private void addDocsRandomly() throws IOException, SolrServerException {
     solrCluster.getSolrClient().commit("collection1_local");
     solrCluster.getSolrClient().commit("collection2_local");
   }
+
+  @Test
+  @SuppressWarnings("unchecked")
+  public void testDisableDistribStats() throws Exception {
+
+    // single collection with implicit router
+    createCollection(COLLECTION, "conf1", ImplicitDocRouter.NAME);
+    SolrClient client = solrCluster.getSolrClient();
+
+    SolrInputDocument doc = new SolrInputDocument();
+    doc.setField("id", "1");
+    doc.setField("cat", "tv");
+    doc.addField(ShardParams._ROUTE_, "a");
+    client.add(COLLECTION, doc);
+
+    doc = new SolrInputDocument();
+    doc.setField("id", "2");
+    doc.setField("cat", "ipad");
+    doc.addField(ShardParams._ROUTE_, "b");
+    client.add(COLLECTION, doc);
+
+    // distributed stats implicitly enabled by default
+    SolrQuery query =
+        new SolrQuery(
+            "q", "*:*",
+            "fl", "id",
+            "fq", "{!terms f=id}1,2",
+            "debug", "track");
+    QueryResponse rsp = client.query(COLLECTION, query);
+    NamedList<Object> track = (NamedList<Object>) rsp.getDebugMap().get("track");
+    assertNotNull(track);
+    assertNotNull(track.get("PARSE_QUERY"));
+
+    // distributed stats explicitly disabled
+    query.set(CommonParams.DISABLE_DISTRIB_STATS, "true");
+    rsp = client.query(COLLECTION, query);
+    track = (NamedList<Object>) rsp.getDebugMap().get("track");
+    assertNotNull(track);
+    assertNull(track.get("PARSE_QUERY"));
+
+    // distributed stats explicitly enabled
+    query.set(CommonParams.DISABLE_DISTRIB_STATS, "false");
+    rsp = client.query(COLLECTION, query);
+    track = (NamedList<Object>) rsp.getDebugMap().get("track");
+    assertNotNull(track);
+    assertNotNull(track.get("PARSE_QUERY"));
```

Comment:
I understand the testing is hard. I barely understand the existing test logic. I think, we can figure out how to extend the existing one.

Reply:
adjusted test query

Reply:
Pardon. I don't see actual testing/asserting logic/clue. How we can make sure the this flag changes anything? May we just put a few docs unevenly and assert that score is different with invoking this flag true and false?